### ISSUE TITLE
stateful_lstm different results using different batch_size

### ISSUE THREAD

**Issue by yurkor** [fchollet/keras#3073]
https://github.com/fchollet/keras/blob/master/examples/stateful_lstm.py
I changed batch_size from 25 to 250, predicted results were changed
print("loss", model.evaluate(cos, expected_output, batch_size=batch_size, verbose=2))
for batch=25 loss 27.7577994211
for batch=250 loss 56.7893032081

on plot new prediction are squashed with max abs y around 50 instead of 100.

**Comment by sadeghmir** [comment]
Some variation is expected. When you change batch size, the number of updates per pass over the data changes. Depending on the update regime, the overall converge speed can change. Try giving it more epochs for the higher batch size and see if you get better results. In any case, can anyone shed some light on the way updates are handled in stateful mode across a batch? Specifically, are they synchronous?

**Comment by MrGroot97** [comment]
> Some variation is expected. When you change batch size, the number of updates per pass over the data changes. Depending on the update regime, the overall converge speed can change. Try giving it more epochs for the higher batch size and see if you get better results. In any case, can anyone shed some light on the way updates are handled in stateful mode across a batch? Specifically, are they synchronous?

is variation expected in the latest TensorFlow(>2.3) also?
